src/Microsoft.DotNet.Darc/src/Darc/Operations/GatherDropOperation.cs

```diff
@@ -876,7 +876,7 @@ private async Task<(bool success, bool anyShipping, ConcurrentBag<DownloadedAsse
             bool success = true;
             var downloaded = new ConcurrentBag<DownloadedAsset>();
             bool anyShipping = false;
-            using (HttpClient client = new HttpClient(new HttpClientHandler { CheckCertificateRevocationList = true }))
+            using (HttpClient client = new HttpClient(new HttpClientHandler { CheckCertificateRevocationList = true }) { Timeout = TimeSpan.FromMinutes(5) })
```

Comment:
Doesn't the http client timeout have to be at least as long as the download timeout, or is that used in a different way?

Reply:
I think the issue is that it takes very close to 5 minutes to download the whole file, but then we have to stream it to the final location. So if they are both 5 minutes, we still end up timing out, but if the cancellation timeout is 100s longer, that gives us enough wiggle room.